allennlp/data/fields/multilabel_field.py

```diff
@@ -0,0 +1,129 @@
+    one hot encoding for the labels (all zeros, and ones for the labels).
+
+    Parameters
+    ----------
+    labels : ``Set[Union[str, int]]``
+    label_namespace : ``str``, optional (default="labels")
+        The namespace to use for converting label strings into integers.  We map label strings to
+        integers for you (e.g., "entailment" and "contradiction" get converted to 0, 1, ...),
+        and this namespace tells the ``Vocabulary`` object which mapping from strings to integers
+        to use (so "entailment" as a label doesn't get the same integer id as "entailment" as a
+        word).  If you have multiple different label fields in your data, you should make sure you
+        use different namespaces for each one, always using the suffix "labels" (e.g.,
+        "passage_labels" and "question_labels").
+    skip_indexing : ``bool``, optional (default=False)
+        If your labels are 0-indexed integers, you can pass in this flag, and we'll skip the indexing
+        step.  If this is ``False`` and your labels are not strings, this throws a ``ConfigurationError``.
+    """
+    # It is possible that users want to use this field with a namespace which uses OOV/PAD tokens.
+    # This warning will be repeated for every instantiation of this class (i.e for every data
+    # instance), spewing a lot of warnings so this class variable is used to only log a single
+    # warning per namespace.
+    _already_warned_namespaces: Set[str] = set()
+
+    # Dictionary from a vocab namespace to size. Need the vocab size to map labels to a tensor
+    _vocab_size: Dict[str, int] = defaultdict(int)
+
+    def __init__(self,
+                 labels: Set[Union[str, int]],
+                 label_namespace: str = 'labels',
+                 skip_indexing: bool = False) -> None:
+        self.labels = labels
+        self._label_namespace = label_namespace
+        self._label_ids = None
+        self._maybe_warn_for_namespace(label_namespace)
+
+        if skip_indexing:
+            for label in labels:
+                if not isinstance(label, int):
+                    raise ConfigurationError("In order to skip indexing, your labels must be integers. "
+                                             "Found label = {}".format(label))
+                # vocabulary size = largest label id
+                MultiLabelField._vocab_size[self._label_namespace] = \
+                    max(label + 1, MultiLabelField._vocab_size[self._label_namespace])
+
+            self._label_ids = labels
+        else:
+            for label in labels:
+                if not isinstance(label, str):
+                    raise ConfigurationError("MultiLabelFields expects string labels if skip_indexing=False. "
+                                             "Found label: {} with type: {}.".format(label, type(label)))
+
+    def _maybe_warn_for_namespace(self, label_namespace: str) -> None:
```

Comment:
it looks like this is only ever called in the constructor, so I might just put the logic there rather than have a separate function for it? (and then you can just use `label_namespace` instead of `self._label_namespace` which would make it a tiny bit easier to read

Reply:
good point, but how about keeping it as a function just to remain consistent with the code in allennlp/data/fields/label_field.py ?